### PR TITLE
Atchub pe preview  microsoft.hybrid compute 2020 08 15 preview

### PR DESCRIPTION
**Pull request by Atchub**
this PR is to add the Private endpoint scenario for HybridCompute . It includes already reviewed changes from https://github.com/Azure/azure-rest-api-specs/pull/10496
the differences being
1) the API version is changed to Preview
2) privateLinkScopedResources property is added on Machine resource to introduce the Privatelinkscope information on Machine
3) linkedResources list in ScopedResource is changed to a string from an array of strings to simplify the scenario.

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [X ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2020-08-15-preview/HybridCompute.json [hunk 1]
```diff
@@ -0,0 +1,1205 @@
...
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HybridCompute/machines/{name}": {
+      "put": {
+        "tags": [
+          "machines"
+        ],
+        "operationId": "Machines_CreateOrUpdate",
+        "description": "The operation to create or update a hybrid machine resource identity in Azure.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
```
Comment (jianyunt) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2020-08-15-preview/HybridCompute.json:46]: any reasons we go common-types for ApiVersionParameter and SubscriptionParameter again? Our previous had defined them locally.

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2020-08-15-preview/examples/Machines_CreateOrUpdate.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          "errorDetails": null,
+          "displayName": null,
+          "machineFqdn": null,
+          "vmUuid": null,
+          "osSku": null,
+          "domainName": null,
+          "adFqdn": null,
+          "dnsFqdn": null,
+          "privateLinkScopedResources": [
+            "/subscriptions/00000000-1111-2222-3333-444444444444/resourceGroups/MyResourceGroup/providers/Microsoft.HybridCompute/privateLinkScopes/MyPrivateLinkScope/scopedResources/scoped-resource-name"
```
Comment (jianyunt) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2020-08-15-preview/examples/Machines_CreateOrUpdate.json:48]: use camel case, myResourceGroup, myPrivateLinkScope...

**Review by jianyunt — APPROVED** [review]

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2020-08-15-preview/examples/Machines_CreateOrUpdate.json (hunk 2 above)
Comment (Atchub) [reply specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2020-08-15-preview/examples/Machines_CreateOrUpdate.json]: fixed them

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2020-08-15-preview/HybridCompute.json (hunk 1 above)
Comment (Atchub) [reply specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2020-08-15-preview/HybridCompute.json:46]: the previous base has them as common-types, so I updated mine to use the common-types. the tool is complaining as duplicates if one file uses common-types and other the local ones. so made it uniform

**Review by pilor — APPROVED** [review]

**Comment by Atchub** [comment]
@fengzhou-msft , can you please review

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**